### PR TITLE
Feature/final release

### PR DESCRIPTION
**Pull request by GerardoHP**
Final changes on the repo
[x] Add a new endpoint
[x] The endpoint must read items from the CSV concurrently using a worker pool
[x] The endpoint must support the following query params:

```text
type: Only support "odd" or "even"
items: Is an Int and is the amount of valid items you need to display as a response
items_per_workers: Is an Int and is the amount of valid items the worker should append to the response
```

### CODE REVIEW

File: src/pokemon-service/utils/file.go [hunk 1]
```diff
@@ -62,3 +67,60 @@ func (f file) AppendLineToFile(line string) error {
 
 	return nil
 }
+
+func (f file) ReadAllFileConcurrent(even bool, items, items_per_worker int) ([]string, error) {
```
Comment (GerardoHP) [inline src/pokemon-service/utils/file.go]: Here is everything that the reviewer needs to know on the concurrency, everything else in the pr is to allow to the new endpoint to reach the concurrency reading

Comment (CesarCueva) [inline src/pokemon-service/utils/file.go]: Add comment to all the exported stuff

File: src/pokemon-service/utils/file.go [hunk 2]
```diff
@@ -62,3 +67,60 @@ func (f file) AppendLineToFile(line string) error {
...
+	if items_per_worker == 0 {
+		items_per_worker = 1
```
Comment (CesarCueva) [inline src/pokemon-service/utils/file.go]: In Go, the naming convention is camel case

File: src/pokemon-service/utils/file.go [hunk 3]
```diff
@@ -62,3 +67,60 @@ func (f file) AppendLineToFile(line string) error {
...
+	}
+
+	wg := sync.WaitGroup{}
+	ch := make(chan string)
+	mt := sync.RWMutex{}
+	file, err := os.Open(f.filename)
```
Comment (CesarCueva) [inline src/pokemon-service/utils/file.go]: This approach is good and works (don't change it)
I would opt to read the file once, then store the info into slice or map and use it in concurrency logic.

**Review by CesarCueva — APPROVED** [review]

**Comment by CesarCueva** [comment]
You have done an excellent job!

File: src/pokemon-service/utils/file.go (hunk 1 above)
Comment (GerardoHP) [reply src/pokemon-service/utils/file.go]: Done

File: src/pokemon-service/utils/file.go (hunk 2 above)
Comment (GerardoHP) [reply src/pokemon-service/utils/file.go]: Done

File: src/pokemon-service/utils/file.go (hunk 3 above)
Comment (GerardoHP) [reply src/pokemon-service/utils/file.go]: Agree I also thought that in memory and only reading once from the file was a better approach but implies a refactor on how to implement the rep, I'll leave it for another refactor later. Thanks for the tip.

**Pull request merged**